api/envoy/api/v2/route/route_components.proto

```diff
@@ -1125,13 +1125,13 @@ message HedgePolicy {
   // [#not-implemented-hide:]
   type.FractionalPercent additional_request_chance = 2;
 
-  // Indicates that a hedged request should be sent when the per-try timeout
-  // is hit. This will only occur if the retry policy also indicates that a
-  // timed out request should be retried.
-  // Once a timed out request is retried due to per try timeout, the router
-  // filter will ensure that it is not retried again even if the returned
-  // response headers would otherwise be retried according the specified
-  // :ref:`RetryPolicy <envoy_api_msg_route.RetryPolicy>`.
+  // Indicates that a hedged request should be sent when the per-try timeout is hit.
+  // This will only occur if the :ref:`RetryPolicy <envoy_api_msg_route.RetryPolicy>` also indicates that
```

Comment:
I don't think this part is right, retry policy doesn't matter when per try timeout is hit.

Reply:
reference comment: 
```
  // A hedged retry on per try timeout is always retried if there are retries
  // left. NOTE: this is a bit different than non-hedged per try timeouts which
  // are only retried if the applicable retry policy specifies either
  // RETRY_ON_5XX or RETRY_ON_GATEWAY_ERROR. This is because these types of
  // retries are associated with a stream reset which is analogous to a gateway
  // error. When hedging on per try timeout is enabled, however, there is no
  // stream reset.

```

Reply:
Interesting. I was actually working off my experiencing using this feature: I only started seeing (hedged) retries once I set `x-envoy-retry-on`.

Reply:
Looking at this code, it seems that `retries_remaining_` would only get initialized to a non-zero value if `retry_on_` is set:
https://github.com/envoyproxy/envoy/blob/2709b6bfc73d089d823c5cde6c883806ec8f3d8e/source/common/router/retry_state_impl.cc#L121-L127

Thus `shouldRetry` would end up returning `RetryStatus::NoRetryLimitExceeded` for the hedged requests.